lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -2377,6 +2412,22 @@ def validate_alert_rules(self, rules: dict) -> Tuple[bool, str]:
                     ]
                 )
 
+    def validate_scrape_jobs(self, jobs: list) -> bool:
+        """Validate scrape jobs using cos-tool."""
+        if not self.path:
+            logger.debug("`cos-tool` unavailable. Not validating scrape jobs.")
+            return True
+        conf = {"scrape_configs": jobs}
+        with tempfile.NamedTemporaryFile() as tmpfile:
+            with open(tmpfile.name, "w") as f:
+                f.write(yaml.safe_dump(conf))
+            try:
+                self._exec([str(self.path), "validate-config", tmpfile.name])
+            except subprocess.CalledProcessError as e:
+                logger.error("Validating scrape jobs failed: {}".format(e.output))
+                raise
```

Comment:
How about:
```suggestion
                msg = "Validating scrape jobs failed: {}".format(e.output)
                logger.error(msg)
                raise ValueError(msg) from e
```

Reply:
Is there a particular reason for this? Just to change the exception type?

Reply:
To display that same error message as part of the traceback.
Not sure if the original exceptions coming out of _exec would print out the contents of `e.output`.

Reply:
They don't, but they're also not used anywhere other than immediately above this. The exception here is used for flow control, and `CalledProcessError` is explicitly caught (and not logged) as, essentially, the "else" statement.

It is not useful here to raise a different error or even surface the traceback